### PR TITLE
Add MQ User resource

### PR DESCRIPTION
**Pull request by mergenci**
### Description of your changes

Add MQ User resource by consuming https://github.com/upbound/terraform-provider-aws/pull/236.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Manually tested by applying the example manifest.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: config/mq/config.go
Comment (ulucinar) [inline config/mq/config.go:56]: This way the cross-resource reference is generated with a target resource API version set to the source's version, i.e., `v1alpha1`. However, `Broker.v1alpha1.mq` does not exist. We have deprecated `config.Resource.Type` for reasons similar to this in favor of the higher level API `config.Reference.TerraformName`, which automatically handles resolution of the correct kind & version for reference target:
```suggestion
			TerraformName: "aws_mq_broker",
```

**Review by ulucinar — COMMENTED** [review]
Thank you @mergenci for working on the new Terraform & Crossplane resource for managing ActiveMQ users.

File: Makefile
Comment (ulucinar) [inline Makefile:186]: Thanks for resolving this issue :)

File: config/externalname.go
Comment (ulucinar) [inline config/externalname.go:2756]: We are planning to take a second look into this to see whether we can avoid it.

File: examples/mq/v1alpha1/user.yaml
Comment (ulucinar) [inline examples/mq/v1alpha1/user.yaml:49]: Let's move this under to `spec.initProvider`. Please also see: https://github.com/crossplane/upjet/pull/406

File: examples/mq/v1alpha1/user.yaml
Comment (ulucinar) [inline examples/mq/v1alpha1/user.yaml:7]: Let's also add the manual intervention annotation till we consume https://github.com/crossplane/upjet/pull/406 in this provider.

**Review by ulucinar — APPROVED** [review]
Thanks @mergenci, lgtm. Left some comments for you to consider. I think we can merge this PR although we will need https://github.com/crossplane/upjet/pull/406 and https://github.com/crossplane/upjet/pull/407 to prevent a race condition between the `User.mq` and `Broker.mq` controllers.

**Comment by ulucinar** [comment]
Let's wait for #1315 before we merge this...

File: examples/mq/v1alpha1/user.yaml
Comment (mergenci) [reply examples/mq/v1alpha1/user.yaml:50]: Let's postpone this change until after referenced Upjet PR is merged. Proposed change would render the example manifest unusable, because `spec.initProvider` doesn't have `user` field at the moment.

**Comment by ulucinar** [comment]
To prevent the race between the new `User.mq` and `Broker.mq` controllers, we will need to consume https://github.com/crossplane/upjet/pull/406 & https://github.com/crossplane/upjet/pull/407 and do a late-initialization configuration.

**Comment by ulucinar** [comment]
Hi @mergenci,
A local run of `make lint` on the head of this PR's feature branch has succeeded. No need to wait for the linter job if you'd like to continue merging...

**Comment by mergenci** [comment]
Thanks @ulucinar 🙏 This PR has been a wild ride in many ways 🙂

**Pull request merged**
